lib/bloc/input/input_bloc.dart

```diff
@@ -46,47 +50,119 @@ class InputBloc extends Cubit<InputState> {
+      _lightningLinks.linksNotifications.doOnData((event) => _log.v("lightningLinks: $event")),
+      _device.clipboardStream.distinct().skip(1).doOnData((event) => _log.v("clipboardStream: $event")),
+    ]).asyncMap((input) async {
+      _log.v("Incoming input: '$input'");
       // Emit an empty InputState with isLoading to display a loader on UI layer
       emit(InputState(isLoading: true));
       try {
-        final command = await breez_sdk.InputParser().parse(s);
-        _log.v("Parsed command: '${command.protocol}'");
-        switch (command.protocol) {
-          case breez_sdk.InputProtocol.paymentRequest:
-            return handlePaymentRequest(s, command);
-          case breez_sdk.InputProtocol.lnurl:
-            return InputState(protocol: command.protocol, inputData: command.decoded as LNURLParseResult);
-          case breez_sdk.InputProtocol.nodeID:
-            return InputState(protocol: command.protocol, inputData: command.decoded);
-          default:
-            return InputState(isLoading: false);
-        }
+        final parsedInput = await _breezLib.parseInput(input: input);
+        // Todo: Merge these functions w/o sacrificing readability
+        _logParsedInput(parsedInput);
+        return await _handleParsedInput(parsedInput);
       } catch (e) {
         _log.e("Failed to parse input", ex: e);
         return InputState(isLoading: false);
       }
-    }).where((inputState) => inputState != null);
+    });
   }
 
-  Future<InputState?> handlePaymentRequest(String raw, breez_sdk.ParsedInput command) async {
-    final lnInvoice = command.decoded as breez_sdk.LNInvoice;
+  Future<InputState> handlePaymentRequest({required dynamic inputData}) async {
+    late LNInvoice lnInvoice;
+    if (inputData is InputType_Bolt11) {
+      lnInvoice = inputData.invoice;
+    } else if (inputData is InputType_BitcoinAddress) {
+      lnInvoice = await _breezLib.parseInvoice(inputData.address.address);
+    }
+
     NodeState? nodeState = await _breezLib.getNodeState();
     if (nodeState == null || nodeState.id == lnInvoice.payeePubkey) {
-      return null;
+      return InputState(isLoading: false);
     }
     var invoice = Invoice(
-        bolt11: raw,
-        paymentHash: lnInvoice.paymentHash,
-        description: lnInvoice.description ?? "",
-        amountMsat: lnInvoice.amountMsat ?? 0,
-        expiry: lnInvoice.expiry);
+      bolt11: lnInvoice.bolt11,
+      paymentHash: lnInvoice.paymentHash,
+      description: lnInvoice.description ?? "",
+      amountMsat: lnInvoice.amountMsat ?? 0,
+      expiry: lnInvoice.expiry,
+    );
+
+    return InputState(inputType: InputType_Bolt11, inputData: invoice);
+  }
 
-    return InputState(protocol: command.protocol, inputData: invoice);
+  Future<InputState> _handleParsedInput(InputType parsedInput) async {
+    if (parsedInput is InputType_Bolt11 || parsedInput is InputType_BitcoinAddress) {
```

Comment:
Looks like it is better to split. Are we doing anything with bitcoin address in breez mobile?